actionpack/CHANGELOG.md

```diff
@@ -1,3 +1,19 @@
+*   Purpose metadata for signed/encrypted cookies.
+
+    Rails can now thwart attacks that attempt to copy signed/encrypted value
+    of a cookie and use it as the value of another cookie.
+
+    It does so by stashing the cookie-name in the purpose field which is
+    then signed/encrypted along with the cookie value. Then, on a server-side
+    read, we verify the cookie-names and discard any attacked cookies.
+
+    Enable `action_dispatch.use_cookies_with_metadata` to use this feature, which
```

Comment:
We need to mention this config in the configuration guide and possibly in the upgrading guide as well (we could look to what we said in 5.2 about the cookie changes for inspiration).